.github/workflows/vmsdk-test-python.yaml

```diff
@@ -5,28 +5,52 @@ on:
     branches:
       - main
     paths:
-      - 'vmsdk/**/*.py'
       - 'common/**/*.py'
+      - 'setupenv.sh'
+      - '.github/workflows/vmsdk-test-python.yaml'
   pull_request:
     paths:
-      - 'vmsdk/**/*.py'
       - 'common/**/*.py'
+      - 'setupenv.sh'
+      - '.github/workflows/vmsdk-test-python.yaml'
   workflow_dispatch:
 
+env:
+  VMSDK_PYTEST_DIR: 'vmsdk_pytest'
+  VMSDK_DIR: 'cc-trusted-vmsdk'
+
 jobs:
   vmsdk_pytest:
     runs-on: [self-hosted, tdx-guest]
+    defaults:
+      run:
+        working-directory: ${{env.VMSDK_PYTEST_DIR}}
     steps:
-      - uses: actions/checkout@v3
+      - name: Clean up intermediate files
+        continue-on-error: true
+        run: |
+          # Remove the intermediate files that could be left
+          # by previous run with sudo. Otherwise, the checkout
+          # will fail with permission issue.
+          sudo rm -fr ./*
+      - name: Checkout cc-trusted-api repo
+        uses: actions/checkout@v4
+        with:
+          path: ${{env.VMSDK_PYTEST_DIR}}
+      - name: Checkout cc-trusted-vmsdk repo
+        uses: actions/checkout@v4
+        with:
+          repository: cc-api/cc-trusted-vmsdk
+          path: ${{env.VMSDK_PYTEST_DIR}}/${{env.VMSDK_DIR}}
       - name: Run PyTest for VMSDK
         run: |
           set -ex
-          # Set the "PYTHONDONTWRITEBYTECODE" and "no:cacheprovider" to prevent
-          # generated some intermediate files by root. Othwerwise, these
-          # files will fail the action/checkout in the next round of running
-          # due to the permission issue.
-          sudo su -c "source setupenv.sh && \
-                      pushd vmsdk/python/tests && \
-                      export PYTHONDONTWRITEBYTECODE=1 && \
-                      python3 -m pytest -p no:cacheprovider -v test_sdk.py && \
-                      popd"
+          cd ${{env.VMSDK_DIR}}
+          # The trick here is that we need to use the local version
+          # of cctrusted_base instead of the version instaleed by
```

Comment:
typo. installed

Reply:
Fixed. Please review the latest version. Thanks!